### ISSUE TITLE
#[cfg(test)] prevents macro expansion

### ISSUE THREAD

**Issue by JamesHallowell** [intellij-rust/intellij-rust#9059]
Hi,

I'm not 100% sure this is a bug or if I'm misunderstanding something, but the behaviour caught me off guard so I thought I should create an issue.

## Environment

* **IntelliJ Rust plugin version:** 0.4.174.4743-221
* **Rust toolchain version:** 1.62.0 (a8314ef7d 2022-06-27) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1.3 (CL-221.5921.27)
* **Operating system:** macOS 11.2
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description

This problem I'm having is that macro expansion works great for my macro until I use it inside a test, at which point it appears to fail and I get false positive errors in the UI.

I have a macro that uses labelled blocks in the input (i.e. `'hello: { ... }`), but these blocks get removed in the expanded output.

The macro compiles fine on stable, but when I use the macro somewhere behind `#[cfg(test)]` then CLion warns me that "label on block is experimental [E0658]" which suggests that its not seeing that the label on the block doesn't exist in the expanded output. The code continues to compile successfully.

## Steps to reproduce

Here is a minimised example:

Cargo.toml
```toml
[package]
name = "macro_bug"
version = "0.0.0"
edition = "2021"

[lib]
proc-macro = true
```

lib.rs
```rust
use proc_macro::TokenStream;

#[proc_macro_attribute]
pub fn macro_that_expands_to_nothing(_args: TokenStream, _input: TokenStream) -> TokenStream {
    "".parse().unwrap()
}
```

test.rs
```rust
use macro_bug::macro_that_expands_to_nothing;

#[cfg(test)]
#[macro_that_expands_to_nothing]
fn foo() {
    'hello: {

    }
}
```

![macro](https://user-images.githubusercontent.com/5460955/178849107-1a01d03f-7117-4dfe-ba4b-c09b2a0b88c3.gif)

Is this expected behaviour? Am I missing something that would allow macro expansion to work in tests also? Or it a known limitation of macro expansion in its current state?

Thanks!
